src/yaca_lib.rs

```diff
@@ -0,0 +1,161 @@
+use libc::{c_char, c_int, c_void, size_t};
+
+
+#[link(name = "yaca")]
+extern {
+    // crypto
+    pub fn yaca_initialize() -> c_int;
+    pub fn yaca_cleanup();
+    pub fn yaca_free(memory: *mut c_void);
+    pub fn yaca_memcmp(first: *const c_void, second: *const c_void, len: size_t) -> c_int;
+    pub fn yaca_randomize_bytes(data: *mut c_char, data_len: size_t) -> c_int;
+    pub fn yaca_context_set_property(ctx: *mut c_void, property: c_int,
```

Comment:
I get what you're saying. I was aware of this when I wrote that code.
Ignored for now. Those enums will be ints in gcc/clang for eternity. We'll get back to this in the future if yaca will be used more and we'll need this cleaned up. I'd have to write this glue library in yaca itself.

Reply:
Sure. Keeping ```yaca``` as it is is also an option, together with changing those ```c_int```s to ```repr(C) enum```s. A decent bindgen capable of automating this may also exist somewhere.